test/XRP/web-socket-network-client.test.ts

```diff
@@ -176,4 +181,89 @@ describe('WebSocket Tests', function (): void {
       }
     }
   })
+
+  it('getAccountOffers - valid requests', async function (): Promise<void> {
+    this.timeout(timeoutMs)
+
+    const issuedCurrencyClient = IssuedCurrencyClient.issuedCurrencyClientWithEndpoint(
+      rippledGrpcUrl,
+      rippledWebSocketUrl,
+      console.log,
+      XrplNetwork.Test,
+    )
+
+    // GIVEN a valid test address with no offers
+    const xAddress = wallet.getAddress()
+    const classicAddress = XrpUtils.decodeXAddress(xAddress)
+    const address = classicAddress!.address
+
+    // WHEN getAccountOffers is called for that address
+    const accountOfferResponse = await webSocketNetworkClient.getAccountOffers(
+      address,
+    )
+
+    // THEN the request is successfully submitted and received, with no listed offers
+    assert.equal(accountOfferResponse.status, ResponseStatus.success)
+    assert.equal(accountOfferResponse.type, 'response')
+
+    const result = (accountOfferResponse as AccountOffersSuccessfulResponse)
+      .result
+
+    assert.equal(result.account, address)
+    assert.isEmpty(result.offers)
+    this.timeout(timeoutMs)
+
+    // GIVEN a valid test address with an offer
+    const takerGetsIssuedCurrency: IssuedCurrency = {
+      issuer: address,
+      currency: 'FAK',
+      value: '100',
+    }
+    const takerPaysXrp = '50'
+
+    await issuedCurrencyClient.createOffer(
+      wallet,
+      takerGetsIssuedCurrency,
+      takerPaysXrp,
+    )
+
+    // WHEN getAccountOffers is called for that address
+    const accountOfferResponse2 = await webSocketNetworkClient.getAccountOffers(
+      address,
+    )
```

Comment:
there's an opportunity to use `Promise.all` here instead of two successive `await`s:

```javascript
const issuedCurrencyPromise = issuedCurrencyClient.createOffer(
  wallet,
  takerGetsIssuedCurrency,
  takerPaysXrp,
)
const accountOfferPromise = webSocketNetworkClient.getAccountOffers(
  address,
)
const [_issuedCurrencyResponse, accountOfferResponse2] = await Promise.all([issuedCurrencyPromise, accountOfferPromise])
```

Reply:
Does this still apply if I want the second `await` to only run after the first one has finished? Because that's the case here.

Reply:
ah i see